pkg/controllers/mcs/service_export_controller.go

```diff
@@ -403,6 +399,43 @@ func reportEndpointSlice(c client.Client, endpointSlice *unstructured.Unstructur
 	return nil
 }
 
+func getEndpointSliceWorkMeta(c client.Client, ns string, workName string, endpointSlice *unstructured.Unstructured) (metav1.ObjectMeta, error) {
+	existWork := &workv1alpha1.Work{}
+	var err error
+	if err := c.Get(context.Background(), types.NamespacedName{
+		Namespace: ns,
+		Name:      workName,
+	}, existWork); err != nil && !apierrors.IsNotFound(err) {
+		klog.Errorf("Get EndpointSlice work(%s/%s) error:%v", ns, workName, err)
+		return metav1.ObjectMeta{}, err
+	}
+
+	labels := map[string]string{
+		util.ServiceNamespaceLabel: endpointSlice.GetNamespace(),
+		util.ServiceNameLabel:      endpointSlice.GetLabels()[discoveryv1.LabelServiceName],
+		// indicate the Work should be not propagated since it's collected resource.
+		util.PropagationInstruction: util.PropagationInstructionSuppressed,
+		util.ManagedByKarmadaLabel:  util.ManagedByKarmadaLabelValue,
+	}
+	if existWork.Labels == nil || (err != nil && apierrors.IsNotFound(err)) {
+		labels[util.EndpointSliceWorkManagedByLabel] = util.ServiceExportKind
+		workMeta := metav1.ObjectMeta{Name: workName, Namespace: ns, Labels: labels}
+		return workMeta, nil
+	}
+
+	labels = util.DedupeAndMergeLabels(labels, existWork.Labels)
+	if value, ok := existWork.Labels[util.EndpointSliceWorkManagedByLabel]; ok {
+		controllerSet := sets.New[string]()
+		controllerSet.Insert(strings.Split(value, "-")...)
```

Comment:
"-" Symbol is prone to ambiguity. How about replacing it with other symbol?

Reply:
+1

Reply:
How about `.`, the only allowed symbol contains: -, _ and .